providers/aws-ec2/src/main/java/org/jclouds/aws/ec2/domain/Route.java

```diff
@@ -0,0 +1,125 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.aws.ec2.domain;
+
+import javax.annotation.Resource;
+import javax.inject.Named;
+
+import org.jclouds.compute.reference.ComputeServiceConstants;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+import org.jclouds.logging.Logger;
+
+import com.google.auto.value.AutoValue;
+
+/**
+ * A route in an Amazon EC2 Route Table.
+ *
+ * @see <a href="http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_Route.html" >doc</a>
+ */
+@AutoValue
+public abstract class Route {
+
+   @Resource
+   @Named(ComputeServiceConstants.COMPUTE_LOGGER)
+   private static Logger logger = Logger.NULL;
+
+   public enum RouteState {
+
+      /**
+       * An active route.
+       */
+      ACTIVE,
+
+      /**
+       * Indicates that the route's target isn't available (for example, the specified gateway isn't attached
+       * to the VPC, or the specified NAT instance has been terminated).
+       */
+      BLACKHOLE,
+
+      /**
+       * Value supplied was not valid.
+       */
+      UNRECOGNIZED;
+
+      public String value() {
+         return name().toLowerCase();
+      }
+
+      public static RouteState fromValue(String v) {
+         if (v == null || v.isEmpty()) {
+            throw new IllegalArgumentException("Value cannot be null or empty");
+         }
+         try {
+            return valueOf(v.toUpperCase());
+         } catch (IllegalArgumentException e) {
+            logger.warn("Unrecognized value supplied for %s: %s", RouteState.class.getSimpleName(), v);
+            return UNRECOGNIZED;
+         }
+      }
+   }
+
+
+   @Nullable
+   public abstract String destinationCidrBlock();
+
+   @Nullable
+   public abstract String gatewayId();
+
+   @Nullable
+   public abstract RouteState state();
+
+   @Nullable
+   public abstract String origin();
+
+   @SerializedNames({"destinationCidrBlock", "gatewayId", "state"})
+   public static Route create(String destinationCidrBlock, String gatewayId, RouteState state) {
+      return builder()
+         .destinationCidrBlock(destinationCidrBlock)
+         .gatewayId(gatewayId)
+         .state(state)
+         .build();
+   }
+
+   Route() {}
+
+   public static Builder builder() {
+      return new AutoValue_Route.Builder();
+   }
+
+   @AutoValue.Builder
+   public abstract static class Builder {
+      public abstract Builder destinationCidrBlock(String destinationCidrBlock);
+
+      public abstract Builder gatewayId(String gatewayId);
+
+      public abstract Builder state(RouteState state);
+
+      public abstract Builder origin(String origin);
+
+      @Nullable abstract String destinationCidrBlock();
+      @Nullable abstract String gatewayId();
+      @Nullable abstract RouteState state();
+      @Nullable abstract String origin();
```

Comment:
These non-visible accessors are not needed. Remove them. We use them when we need access to the internal builder values in a custom build method to enforce immutability.